analyzers/tests/SonarAnalyzer.Test/Rules/SecureRandomSeedsShouldNotBePredictableTest.cs

```diff
@@ -0,0 +1,154 @@
+                using Org.BouncyCastle.Security;
+                class Testcases
+                {
+                    void Method()
+                    {
+                        var sr = SecureRandom.GetInstance("SHA256PRNG", false);
+                        {{{expression}}};                  // Noncompliant {{Set an unpredictable seed before generating random values.}}
+
+                        sr = SecureRandom.GetInstance("SHA256PRNG", true);
+                        {{{expression}}};                  // Compliant, autoSeed is set to true
+
+                        sr = SecureRandom.GetInstance("SHA256PRNG");
+                        {{{expression}}};                  // Compliant, autoSeed is implicitly set to true
+                    }
+                }
+                """)
+            .Verify();
+
+    [DataRow("new byte[42]")]
+    [DataRow("new byte[] { 1, 2, 3 }")]
+    [DataRow("new byte[] { (byte)'a', (byte)'b', (byte)'c' }")]
+    [DataRow("""Encoding.UTF8.GetBytes("exploding whale")""")]
+    [DataRow("""Encoding.UTF8.GetBytes(SOME_STRING)""")]
+#if NET
+    [DataRow("""Encoding.UTF8.GetBytes("abc", 0, 1)""")]
+#endif
+    [DataRow("""Encoding.UTF8.GetBytes(new char[42])""")]
+    [DataRow("""Encoding.UTF8.GetBytes(new char[] {'1', '2'})""")]
+    [DataRow("""Encoding.UTF8.GetBytes(new char[] {'1', '2'}, 0, 1)""")]
+    [DataRow("""Encoding.UTF8.GetBytes(new char[] {'1', '2'}, 0, 1, null, 42)""")]
+    [DataRow("""Convert.FromBase64String("exploding whale")""")]
+    [DataRow("""Convert.FromBase64String(SOME_STRING)""")]
+    [DataRow("""Convert.FromBase64CharArray(new char[] {'1', '2'}, 0, 42)""")]
+    [DataTestMethod]
+    public void SecureRandomSeedsShouldNotBePredictable_CS_Next_WithSetSeed(string expression) =>
+        builder
+            .AddSnippet($$$"""
+                using System;
+                using System.Text;
+                using Org.BouncyCastle.Security;
+                class Testcases
+                {
+                    const string SOME_STRING = "exploding whale";
+
+                    void StartWithUnpredictable(byte[] whoKnows)
+                    {
+                        var sr = SecureRandom.GetInstance("SHA256PRNG", false);
+
+                        sr.SetSeed(whoKnows);
+                        sr.Next();                  // Compliant, seed is updated with unknown data
+
+                        sr.SetSeed({{{expression}}});
+                        sr.Next();                  // Compliant, seed is updated with non-random data, but it (might) have already been random
+                    }
+
+                    void StartWithPredictable()
+                    {
+                        var sr = SecureRandom.GetInstance("SHA256PRNG", false);
+
+                        var seed = {{{expression}}};
+                        sr.SetSeed(seed);
+                        sr.Next();                  // Noncompliant {{Set an unpredictable seed before generating random values.}}
```

Comment:
Shouldn't the message, in this case when a seed has been set but it's predictable, be "Change this seed value to something unpredictable, or remove the seed?"?

Reply:
Although, if it's too difficult to customize the message, it might not be worth it.

Reply:
I think it's going to be a bit difficult to customize it, and also you can `SetSeed` multiple times, and only once you have to be "unpredictable", as you are changing the generator's state, so I think the message works both ways.